### ISSUE TITLE
Resolve install doc differences

### ISSUE THREAD

**Issue by jodh-intel** [kata-containers/documentation#335]
PR #329 modified a single install doc to use the latest stable branch rather than the master branch.

However, I'm rather confused by it since:

- It is only changing a *single* distros installation instructions.
  - Meaning all the other instructions are now different.

    We have tried *very* hard to keep the docs as similar as possible so this seems strange.

- It's not clear to me that this is the correct fix.
- We're hard coding a version number which is going to go stale fast.
- We now have to remember to manually update this doc for every release.
  - And what about the other docs?

The question is what are we going to do now?

- Revert #329.
- Change all the remaining docs to specify a stable branch somehow?

... or just "bite the bullet" and [fork the doc repo](https://github.com/kata-containers/documentation/issues/246).

**Comment by grahamwhaley** [comment]
Good call @jodh-intel . I think at the higher level, as you say:
 - hard wiring a version into the master is probably wrong
 - forking stable docs branches may be the right thing to do (we have some big changes in flight, maybe now is the time...), and if we do that then hard wiring (or somehow determining, possibly using some `git` magic?) the correct version to use seems to be the right thing.

In this specific instance, I think the hard wiring was to work around a broken package issue with that one version/distro - in which case the minimum we should do in parallel to it is open an Issue to track that breakage and later revert the hard wiring.
/cc @chavafg

**Comment by gnawux** [comment]
> We now have to remember to manually update this doc for every release.

I hate the taste...

**Comment by gnawux** [comment]
> forking stable docs branches may be the right thing to do (we have some big changes in flight, maybe now is the time...), and if we do that then hard wiring (or somehow determining, possibly using some git magic?) the correct version to use seems to be the right thing.

Documents for each stable releases? I have to say it might be too big right now.

**Comment by egernst** [comment]
@jodh-intel thanks for opening this -- I intended to to open this with the prior merge.

The Ubuntu instructions were updated since our install was 100% broken on master last week, and a few people were thus reporting issues as they began to 'start playing with' Kata.  I wanted to fix in the short term.

Long term, it probably makes sense to use something like 'latest stable' here, imo.  I really like how K8S does this: 

```
curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/darwin/amd64/kubectl
```

**Comment by jodh-intel** [comment]
@egernst - understood.

@gnawux - I don't think branching would create a lot of extra work here? I was imagining a lightweight process like the following after we've created a new release:

- Create a new `stable-${version}` branch in the docs repo.
- Raise a PR that changes `master` to `stable-${version}` in all the install docs in that branch.
- Wait for the CI to test the modified installation docs (using the latest OBS-created  `stable-${version}` binaries).
- Merge the PR.
- Announce the new release pointing users at the docs at:

  https://github.com/kata-containers/documentation/tree/stable-${version}/install/

I think all of the above could be scripted too.

Once the release is done, there would be no reason to modify the version-specific docs unless we ended up having to deal with difficult scenarios before that stable release ended up being deprecated.
